src/lib/utils/hooks/SynapseAPI/forum/useForum.ts

```diff
@@ -317,3 +317,23 @@ export function usePutReply(
     },
   )
 }
+
+export function useDeleteReply(
+  options?: UseMutationOptions<void, SynapseClientError, string>,
+) {
+  const queryClient = useQueryClient()
+  const { accessToken } = useSynapseContext()
+
+  return useMutation<void, SynapseClientError, string>(
+    (replyId: string) => SynapseClient.deleteReply(accessToken, replyId),
+    {
+      ...options,
+      onSuccess: async (updatedReply, replyId, ctx) => {
+        await queryClient.invalidateQueries(['thread'])
```

Comment:
Can you pass the thread ID just to get a more specific cache key invalidation?
```suggestion
        await queryClient.invalidateQueries(['thread', threadId])
```